packages/common/src/services/Jupiter.ts

```diff
@@ -23,8 +28,59 @@ export const parseJupiterInstruction = (instruction: Instruction) => {
   })
 }
 
+let jupiterClient: ReturnType<typeof createJupiterApiClient> | null = null
+
+export const getJupiterClient = () => {
+  if (!jupiterClient) {
+    jupiterClient = createJupiterApiClient()
+  }
+  return jupiterClient
+}
+
+// Legacy instance for backward compatibility
 export const jupiterInstance = createJupiterApiClient()
 
+export type JupiterQuoteParams = {
+  inputTokenSymbol: JupiterTokenSymbol
+  outputTokenSymbol: JupiterTokenSymbol
+  inputAmount: number
+  slippageBps: number
+  swapMode?: 'ExactIn' | 'ExactOut'
+  onlyDirectRoutes?: boolean
+}
+
+// Add support for mint-based parameters for the useSwapTokens hook
+export type JupiterMintQuoteParams = {
```

Comment:
we can't grab these types from jupiter somehow?

Reply:
we're doing some custom stuff here